diff --git runbot/runbot.py runbot/runbot.py
index 2b082c7..04bf199 100644
--- runbot/runbot.py
+++ runbot/runbot.py
@@ -275,8 +275,11 @@ class runbot_repo(osv.osv):
             run(['git', 'clone', '--bare', repo.name, repo.path])
 
         # check for mode == hook
-        fetch_time = os.path.getmtime(os.path.join(repo.path, 'FETCH_HEAD'))
-        if repo.mode == 'hook' and repo.hook_time and dt2time(repo.hook_time) < fetch_time:
+	fname_fetch_head = os.path.join(repo.path, 'FETCH_HEAD')
+	fetch_time = None
+	if os.path.isfile(fname_fetch_head):
+            fetch_time = os.path.getmtime(fname_fetch_head)
+        if fetch_time and repo.mode == 'hook' and repo.hook_time and dt2time(repo.hook_time) < fetch_time:
             t0 = time.time()
             _logger.debug('repo %s skip hook fetch fetch_time: %ss ago hook_time: %ss ago', repo.name, int(t0 - fetch_time), int(t0 - dt2time(repo.hook_time)))
             return
